### PR TITLE
Add package & install script for npm

### PR DESCRIPTION
**Pull request by jskrzypek**
This PR adds the needed files to allow users to install via npm, as discussed in #31

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "sentry-cli",
+  "version": "0.25.0",
+  "description": "A command line utility to work with Sentry. https://docs.sentry.io/hosted/learn/cli/",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "preinstall": "INSTALL_DIR=./bin ./bin/install-sentry-cli"
```
Comment (mattrobenolt) [inline package.json]: Is there a way that we can just leverage our existing install script without duplicating it in two places?

If there are changes to name to the script to accommodate working for both, I think we'd be more inclined to support that, in one place, than a copy/paste in two places where one that we never use will absolutely break at some point in the future. :)

Comment (jskrzypek) [inline package.json]: I couldn't see where you had the source for the script​ in the repo, which is why I copied it. If its in a different repo, you could just as eaily put the `package.json`, `bin/`, & `.npmignore` files there and that way you'd just bump the version in the package & get-cli script simultaneously in your build process.

The other alternative (which seems hackier to me, but may be easier for you guys to support) is to change the `preinstall` npm script to be 
```json
"preinstall": "INSTALL_DIR=./bin curl -sL https://sentry.io/get-cli/"
```

Comment (jskrzypek) [inline package.json]: Let me know which path you prefer to take and i can modify/close the PR as needed

File: package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "cli"
+  ],
+  "bugs": {
+    "url": "https://github.com/getsentry/sentry-cli/issues"
+  },
+  "bin": {
+    "sentry-cli": "bin/sentry-cli"
+  },
+  "homepage": "https://docs.sentry.io/hosted/learn/cli/",
+  "license": ""
```
Comment (mitsuhiko) [inline package.json]: The license of sentry-cli atm is a 3 clause BSD. So that should match it.

File: package.json (hunk 1 above)
Comment (mitsuhiko) [inline package.json]: I'm okay with having the duplicated script here but we should strip it down to the bare minimum. In particular the sudo stuff does not belong there any more and we should depend on `npm_package_version`.

**Comment by mitsuhiko** [comment]
I'm happy to merge this if my above comments are addressed I think.

**Comment by mitsuhiko** [comment]
(Though I do wonder if there is a better way to download this than curl. Because what about windows users?)

**Comment by jskrzypek** [comment]
@mitsuhiko I was also wondering about the portability of curl, but I noticed that's what you're doing inside the install script you are serving at `https://sentry.io/get-cli/` so I stuck with it.

**Comment by mitsuhiko** [comment]
Yeah, just that never supported windows and is documented as such :)

**Comment by mattrobenolt** [comment]
Throwing in my two cents: since we're operating under such a controlled environment and the install script is so minimal, can we just port it to Node and let Node download and store the binary? Seems like we'd at least gain compatibility with platforms the original bash script is not compatible with. That would at least be a win for us, imo.

**Comment by jskrzypek** [comment]
Well, it has the `Windows_i686` link... 😄 

@mattrobenolt You beat me to the suggestion.

An example of something like that can be seen here, where we're using this script internally to make sure we can get wkhtmltopdf on heroku:
https://github.com/Yoobic/wkhtmltopdf/blob/master/wkhtmltopdf.js

I need to check, but we may need to do write it as pure node without the dependency on wget-improved in order for it to run during "preinstall"

**Comment by jskrzypek** [comment]
Ok so I tried it and I can't use the same `wget-improved` library in `preinstall` because it hasn't been installed yet (*duh*).
I'm working on a version that will use the `https` builtin module which is a bit of a headache and I need to handle the 302 redirects manually, but it should be fine. I should have something working by tomorrow.

**Comment by mitsuhiko** [comment]
Awesome! 👍

**Comment by jskrzypek** [comment]
Sorry for the delay, was kind of busy with work the past couple days and I didn't have time to get back to this. 

The new script runs, downloads a functional binary, and checks that the output of `<downloaded_sentry_cli> --version` matches the version in the `package.json` (which is mostly a check that ensures the binary executes correctly).

**Comment by jskrzypek** [comment]
I realized you guys had released a new version, so I rebased & bumped the package.json version to 0.27.1. I suppose you'll need to add the `npm version` & `publish` commands to your build process.

**Comment by mitsuhiko** [comment]
Awesome. Thanks so much. I will make sure with the next release (today or tomorrow) this gets merged and pushed to npm.

**Comment by jskrzypek** [comment]
Great :)

**Comment by ghaiat** [comment]
hello
Any update on this merge?
Cheers

**Comment by mitsuhiko** [comment]
It will land with the next release of sentry-cli

Regards,
Armin

**Comment by jskrzypek** [comment]
So I only just bothered to check that the sentry-cli name is available on
npm... Apparently it isn't :( but i guess you guys will figure out what to
do with that.

For now we're just going to point to the GitHub repo

**Comment by mitsuhiko** [comment]
It's called `sentry-cli-binary` now and I just added it to the docs as well. https://docs.sentry.io/learn/cli/#installation-via-npm

**Pull request merged**